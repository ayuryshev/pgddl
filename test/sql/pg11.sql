\pset null _null_
\pset format unaligned

SET client_min_messages = warning;
SET ROLE postgres;

CREATE TABLE customers(cust_id bigint NOT NULL,cust_name varchar(32) NOT NULL,cust_address text,
cust_country text) PARTITION BY LIST(cust_country);
CREATE TABLE customer_ind PARTITION OF customers FOR VALUES IN ('ind');
CREATE TABLE customer_jap PARTITION OF customers FOR VALUES IN ('jap');
CREATE TABLE customers_def PARTITION OF customers DEFAULT;
INSERT INTO customers VALUES (2039,'Puja','Hyderabad','ind');
SELECT tableoid::regclass,* FROM customers;
SELECT * FROM customer_ind;
UPDATE customers SET cust_country ='jap' WHERE cust_id=2039;
SELECT * FROM customer_jap;

select ddlx_script('customers');
select ddlx_script('customer_jap');

-- test hash partitioning

-- test procedures
