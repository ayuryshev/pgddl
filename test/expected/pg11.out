\pset null _null_
\pset format unaligned
SET client_min_messages = warning;
SET ROLE postgres;
CREATE TABLE customers(cust_id bigint NOT NULL,cust_name varchar(32) NOT NULL,cust_address text,
cust_country text) PARTITION BY LIST(cust_country);
CREATE TABLE customer_ind PARTITION OF customers FOR VALUES IN ('ind');
CREATE TABLE customer_jap PARTITION OF customers FOR VALUES IN ('jap');
CREATE TABLE customers_def PARTITION OF customers DEFAULT;
INSERT INTO customers VALUES (2039,'Puja','Hyderabad','ind');
SELECT tableoid::regclass,* FROM customers;
tableoid|cust_id|cust_name|cust_address|cust_country
customer_ind|2039|Puja|Hyderabad|ind
(1 row)
SELECT * FROM customer_ind;
cust_id|cust_name|cust_address|cust_country
2039|Puja|Hyderabad|ind
(1 row)
UPDATE customers SET cust_country ='jap' WHERE cust_id=2039;
SELECT * FROM customer_jap;
cust_id|cust_name|cust_address|cust_country
2039|Puja|Hyderabad|jap
(1 row)
select ddlx_script('customers');
ddlx_script
BEGIN;

/*
DROP TABLE customers;
*/

-- Type: TABLE ; Name: customers; Owner: postgres

CREATE TABLE customers (
    cust_id bigint NOT NULL,
    cust_name character varying(32) NOT NULL,
    cust_address text,
    cust_country text
)  PARTITION BY LIST (cust_country)
;

COMMENT ON TABLE customers IS NULL;

ALTER TABLE customers OWNER TO postgres;

END;

(1 row)
select ddlx_script('customer_jap');
ddlx_script
BEGIN;

/*
DROP TABLE customer_jap;
*/

-- Type: TABLE ; Name: customer_jap; Owner: postgres

CREATE TABLE customer_jap (

) INHERITS(customers)
;

COMMENT ON TABLE customer_jap IS NULL;

ALTER TABLE customer_jap OWNER TO postgres;

END;

(1 row)
-- test hash partitioning
-- test procedures
